lib/src/test/java/io/ably/lib/test/realtime/RealtimeMessageTest.java

```diff
@@ -825,27 +825,29 @@ private void expectDataToMatch(MessagesEncodingDataItem fixtureMessage, Message
 	public void reject_invalid_message_data() throws AblyException {
 		HashMap<String, Integer> data = new HashMap<String, Integer>();
 		Message message = new Message("event", data);
-		final Log.LogHandler logHandler = Log.handler;
-		try {
-			final ArrayList<LogLine> capturedLog = new ArrayList<>();
-			Log.setHandler(new Log.LogHandler() {
-				@Override
-				public void println(int severity, String tag, String msg, Throwable tr) {
-					capturedLog.add(new LogLine(severity, tag, msg, tr));
-				}
-			});
+		Log.LogHandler originalLogHandler = Log.handler;
+		int originalLogLevel = Log.level;
+		Log.setLevel(Log.DEBUG);
+		final ArrayList<LogLine> capturedLog = new ArrayList<>();
+		Log.setHandler(new Log.LogHandler() {
+			@Override
+			public void println(int severity, String tag, String msg, Throwable tr) {
+				capturedLog.add(new LogLine(severity, tag, msg, tr));
+			}
+		});
 
+		try {
 			message.encode(null);
-
+		} catch (AblyException e) {
```

Comment:
I think there should also be a catch block for the case that a `Throwable` is thrown that isn't an `AblyException` and fail the test.

Reply:
Done https://github.com/ably/ably-java/pull/385/commits/69995806b33671777dd68e989a2aa3ef56e1994d